tools/validation_files.py

```diff
@@ -1,46 +1,53 @@
-from auxiliaries.pipeline_auxiliaries import load_table_to_dict
+from auxiliaries.pipeline_auxiliaries import load_table_to_dict, sample2bc_schema
 import os
 import re
+import json
+import jsonschema
+
 
 def is_same_samples(samples2bc_path, barcode2samples_path):
     # verify that file sample2biologicalcondition and file barcode2sample have the same samples.
     samples_from_sample2bc = sorted(load_table_to_dict(samples2bc_path,'Samples {} belongs to more than one bc!!').keys())
     samples_from_barcode2sample = sorted(load_table_to_dict(barcode2samples_path,'Barcode {} belongs to more than one sample!!').values())
-    return samples_from_sample2bc == samples_from_barcode2sample
+    if not samples_from_sample2bc == samples_from_barcode2sample:
+        print(f'The files: {samples2bc_path} and {barcode2samples_path} not contain the same samples!!')
+        return False
+    return True
 
-def is_valid_structure_file(file_path):
+def is_valid_txt_file(file_path):
     # verify the structure of the file -  should look like: word\tword\n
     regex = '\w*\t\w*\n'
     for data in  open(file_path, "rb"):
         line = data.decode("utf-8")
         m = re.match(regex, line)
         if m == None:
+            print(f'The structure of file name {file_path} is not valid')
             return False
     return True        
 
-def is_input_files_valid(samples2bc_path, barcode2samples_path):
-    samples2bc_valid = False
-    if samples2bc_path:
-        filename, file_extension = os.path.splitext(samples2bc_path)
-        samples2bc_valid = is_valid_structure_file(samples2bc_path) if file_extension=='.txt' else True
-        if samples2bc_valid: 
-            print(f'The structure of file name {samples2bc_path} is valid')
-        else:
-            print(f'The structure of file name {samples2bc_path} is not valid')   
-    barcode2samples_valid = is_valid_structure_file(barcode2samples_path) if barcode2samples_path else False
-    if barcode2samples_valid: 
-            print(f'The structure of file name {barcode2samples_path} is valid')
-    else:
-            print(f'The structure of file name {barcode2samples_path} is not valid')   
+def is_valid_Json_file(json_path, schema):
+    json_data = json.load(open(json_path))
+    try:
+        jsonschema.validate(instance=json_data, schema=schema)
+    except jsonschema.exceptions.ValidationError as err:
+        print(f'The structure of file name {json_path} is not valid')
+        return False
+    return True
 
-    if samples2bc_path and barcode2samples_path:
-        if samples2bc_valid and barcode2samples_valid:
-            same_samples = is_same_samples(samples2bc_path, barcode2samples_path)
-            if not same_samples:
-                print(f'The files: {samples2bc_path} and {barcode2samples_path} not contain the same samples!!')
-                return False
-        else:
-            False
+def is_input_files_valid(samplename2biologicalcondition_path, barcode2samplename_path):
```

Comment:
Please refactor the method it is not clear enough and prone to error.
For example file can invalid but you still test if same samples.
Hard to follow what happens if only file is supplied.
Data is loaded multiple times (validation and same sample).
Return condition doesn't follow best practice where validation functions return a single True value at the end and False value in conditions.

The following is the psuedo code for the functions:
* Set validation flags to **True**
* If bar2sam path is defined
  * Try to load data, if fails set flag to false, log and skip next line
  * Validate data, if invalid set flag to false and log
* If sam2bc path is defined, do the same as above (update the other flag)
* If any flag is false, return False (no reason to check containment)
* if both path exists and not is_same_sample: return False
* return True